packages/peregrine/lib/talons/Header/useStoreSwitcher.js

```diff
@@ -0,0 +1,101 @@
+import { useQuery } from '@apollo/client';
+import { useCallback, useMemo } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useDropdown } from '@magento/peregrine/lib/hooks/useDropdown';
+import { BrowserPersistence } from '@magento/peregrine/lib/util';
+
+const storage = new BrowserPersistence();
+
+const mapAvailableOptions = (rawConfigData, rawAvailableStoresData) => {
+    const { code } = rawConfigData;
+    const availableOptions = {};
+
+    rawAvailableStoresData.forEach(store => {
+        availableOptions[store.code] = {
```

Comment:
If you're going to reduce an iterable (`rawAvailableStoresData` here) to a keyed object, you should consider whether the consumer will

- **know** or **not know** the keys
- **look up** values or **iterate over** entries

In this case, the consumer (component) will **not know** the keys and will **iterate over** entries to create elements, so it'd be better to reduce to a `Map` than a plain object.

```es6
const getAvailableOptions = (config, stores) => {
  const { code: configCode } = config

  const options = stores.reduce(
    (map, store) => {
      const {
        code,
        default_display_currency_code: currency,
        locale,
        store_name: storeName
      } = store

      const isCurrent = code === configCode
      const option = { currency, isCurrent, locale, storeName }

      return map.set(code, option)
    },
    new Map()
  )
}
```

Also, the option object here should probably use `isCurrent` instead of `is_current`, to match the casing of `storeName`.

Reply:
Done, thank you